src/rust/ide/view/graph-editor/src/lib.rs

```diff
@@ -851,6 +862,10 @@ impl Edges {
 
 
 
+#[derive(Debug,Clone,CloneRef,Default)]
+struct Visualisations {
+    selected : SharedHashSet<NodeId>,
+}
```

Comment:
This suggests that more than one visualization can be selected, right?

Reply:
The problem leading to this is, that the visualizations inform us about their selection status, but we cannot know the order of FRP events we get. So, we might get, for example, "vis A select" "vis B" select" "vis A deselect" order of events, and we might want to keep track of this intermediary state. We could just have a single Option here, but this might be more useful.

Reply:
Please add EXTENSIVE docs here - descirbing the architecture, the bottom up events propagation, that this set will contain a single value normally etc. This is very not obvious nad it sohuld be documented :)